packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +829,385 @@ def to_json(self):
+        if isinstance(json_request[i], list) and len(json_request[i])>1:
+                inner_model.append(Token(" ", TokenKind.Newline))
+                inner_model.append(
+                                Token(" " * (indent * 4), TokenKind.Whitespace)
+                            )
+                inner_model.append(Token("}[];", TokenKind.Comment))
+        elif isinstance(json_request[i],list) and isinstance(json_request[i][0],dict):
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("}[];", TokenKind.Comment))
+        elif len(json_request[i])>1:
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+    elif isinstance(json_request[i], list) and indent > 4:
+        if isinstance(json_request[i][0], dict) :
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "}[];", None)
+        if len(json_request[i])==1:
+            pass
+        else:
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "}[];", None)
+    else:
+        if indent == 5 and inner_model:
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+            if self.inner_model != inner_model:
+                self.inner_model +=inner_model
+            else: self.inner_model = inner_model
+            inner_model = []
+        elif inner_model:
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+        else:
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "};", None)
+    return inner_model
+
+def list_format(self, json_request, yaml, not_first, indent, name, inner_model):
+    no_list = True
+    if isinstance(json_request, list):
+        for i in range(0, len(json_request)):
+            if isinstance(json_request[i], str):
+                index = json_request[i].find("(optional)")
+                param = json_request[i].split()
+                if len(param) >= 2:
+                    param[0] = format_param_type(param[0])
+                    if index != -1:
+                        optional = "?"
+                    json_request[i] = optional + " : " + param[0] + "[];"
```

Comment:
same comment for `optional` here, make sure it's defined as an empty string before the if statement otherwise you could hit a `Local variable 'optional' might be referenced before assignment`